analysis/priestdiscipline/src/modules/spells/Penance.js

```diff
@@ -7,33 +7,16 @@ import BoringSpellValue from 'parser/ui/BoringSpellValue';
 import Statistic from 'parser/ui/Statistic';
 import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
 
-const PENANCE_MINIMUM_RECAST_TIME = 3500; // Minimum duration from one Penance to Another
-
 class Penance extends Analyzer {
-  get missedBolts() {
-    return [...this.eventGrouper].reduce((missedBolts, cast) => {
-      cast.length > 3 ? (this._boltCount = 6) : (this._boltCount = 3); // checking for Penitent one legendary
-      return missedBolts + (this._boltCount - cast.length);
-    }, 0);
-  }
-
-  get casts() {
-    return [...this.eventGrouper].length;
-  }
-
-  get currentBoltNumber() {
-    return [...this.eventGrouper].slice(-1)[0].length - 1; // -1 here for legacy code
-  }
-
-  _boltCount = 3;
-  hits = 0;
-  eventGrouper = new EventGrouper(PENANCE_MINIMUM_RECAST_TIME);
+  _expectedBoltCount = 0;
+  _boltCount = 0;
+  _missedBolts = 0;
```

Comment:
I'd suggest having a `totalExpectedBolts` which would be the total number of bolts expected on a fight and then a `totalBoltHits` which would be the actual bolt hits counted in a log
This means we can derive the total missed bolts by doing `totalExpectedBolts - totalBoltHits`